src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1106,6 +1125,35 @@ void GroupDataProviderImpl::EndpointIteratorImpl::Release()
     mProvider.mEndpointIterators.ReleaseObject(this);
 }
 
+CHIP_ERROR GroupDataProviderImpl::RemoveEndpoints(chip::FabricIndex fabric_index, chip::GroupId group_id)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INTERNAL);
+
+    FabricData fabric(fabric_index);
+    GroupData group;
+
+    VerifyOrReturnError(CHIP_NO_ERROR == fabric.Load(mStorage), CHIP_ERROR_INVALID_FABRIC_ID);
+    VerifyOrReturnError(group.Find(mStorage, fabric, group_id), CHIP_ERROR_KEY_NOT_FOUND);
+
+    EndpointData endpoint(fabric_index, group.id, group.first_endpoint);
+    size_t endpoint_index = 0;
+    while (endpoint_index < group.endpoint_count)
+    {
+        if (CHIP_NO_ERROR != endpoint.Load(mStorage))
```

Comment:
could you explain the logic here in a coment?

 - why is an error ok? I.e. why do we return CHIP_NO_ERROR even though some load failed. Why does a load failure of index N mean that we never load/delete everything beyond N+1? 
 
 Ideally I imagine we should check the error actual value if it is some form  of "does not exist" but log any other kind of errors at a minimum. This breaks silently on the first error.

Reply:
The idea behind this condition is to try to recover in case of error. Is the reading fails, the endpoint list is still emptied, even if some entries become orphan. However, you're correct in that the error will be completely silent. Since this condition is not expected to happen under normal operation, I have replace the condition with a verification, and the error, if ever happen, will be reported.

Reply:
Re-opened because latest commits do not seem to show this update. Please resolve once you push again (assuming  just missing a push, keeping this as a marker)